src/couch/src/couch_db_updater.erl

```diff
@@ -1086,8 +1087,14 @@ copy_docs(Db, #db{fd = DestFd} = NewDb, MixedInfos, Retry) ->
         {NewRevTree, FinalAcc} = couch_key_tree:mapfold(fun
             (_Rev, #leaf{ptr=Sp}=Leaf, leaf, SizesAcc) ->
                 {Body, AttInfos} = copy_doc_attachments(Db, Sp, DestFd),
+                EJsonBody = case is_binary(Body) of
```

Comment:
I see why we need to do this for existing shards that may have an incorrect value however I'm not a huge fan of this being a permanent change and not just around for an upgrade. Would be nice if we could figure out how to special case this so it only runs the decompression when we need to upgrade.